crates/bevy_ecs/macros/src/lib.rs

```diff
@@ -90,6 +101,36 @@ pub fn derive_bundle(input: TokenStream) -> TokenStream {
         Err(e) => return e.into_compile_error().into(),
     };
 
+    let mut field_kind = vec![];
```

Comment:
I believe, since we have `named_fields` and `field_kind` will always have exactly the same number of elements, we should use
```suggestion
    let mut field_kind = Vec::with_capacity(named_fields.len());
```